TechnicalTask-lvl1/Common/NetworkMonitor.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  NetworkMonitor.swift
+//  TechnicalTask-lvl1
+//
+//  Created by User on 28/11/2024.
+//
+
+import Combine
+import Reachability
+
+final class NetworkMonitor {
+    
+    static let shared = NetworkMonitor()
+    private let connectionStatusRelay = PassthroughSubject<Bool, Never>()
```

Comment:
Не стоит тащить в `Combine` нейминг из `Rx`
```suggestion
    private let connectionStatusSubject = PassthroughSubject<Bool, Never>()
```